### PR TITLE
Core documentation - generate data type docs programmatically

### LINKED ISSUE #2297: Website is missing docs for standard library templates
The standard library includes some templates like `OptionalProperties<>` that are described in any of the website docs.

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #2297.

### CODE REVIEW

File: docs/standard-library/built-in-data-types.md [hunk 1]
```diff
@@ -4,4 +4,342 @@ toc_min_heading_level: 2
 toc_max_heading_level: 3
 ---
 
-See [Language Basics: Built-in types](../language-basics/built-in-types)
+# Data types
```
Comment (timotheeguerin) [inline docs/standard-library/built-in-data-types.md:7]: This feels liek this has a bit of duplication with the `built-in-types` wondering if we should not omit all scalars and maybe `Array`

Comment (tjprescott) [reply docs/standard-library/built-in-data-types.md:7]: It's supposed to. This is the autogenerated reference docs, rather than the manually written ones.

Comment (timotheeguerin) [reply docs/standard-library/built-in-data-types.md:6]: I guess was removing them from the auto generation but maybe it’s fine to have the duplicate olwith maybe this one having more detailed doc

File: .prettierignore [hunk 2]
```diff
@@ -36,6 +36,7 @@ docs/**/reference/
 docs/standard-library/built-in-decorators.md
 packages/website/versioned_docs/**/reference/
 packages/website/versioned_docs/**/standard-library/built-in-decorators.md
+packages/website/versioned_docs/**/standard-library/built-in-data-types.md
```
Comment (timotheeguerin) [inline .prettierignore:39]: Also need to exclude the docs/ folder one

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**